### PR TITLE
WRN-1373: Migrated Header and Panels unit tests to testing library.

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Migrated Header and Panels unit tests to testing library.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-1373

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

File: Panels/tests/Header-specs.js [hunk 1]
```diff
@@ -1,66 +1,54 @@
...
-			<Header type="wizard"><title>Wizard Header</title></Header>
-		);
+		render(<Header data-testid="header" type="wizard"><title>Wizard Header</title></Header>);
 
 		const expected = css.wizard;
-		const actual = subject.find(`.${css.header}`).first().prop('className');
+		const header = screen.getByTestId('header');
 
-		expect(actual).toContain(expected);
+		expect(header).toHaveClass(expected);
```
Comment (alexandrumorariu) [inline Panels/tests/Header-specs.js:25]: Remove this line.

File: Panels/tests/Header-specs.js [hunk 2]
```diff
@@ -1,66 +1,54 @@
...
-			<Header type="compact"><title>Compact Header</title></Header>
-		);
+		render(<Header data-testid="header" type="compact"><title>Compact Header</title></Header>);
 
 		const expected = css.compact;
-		const actual = subject.find(`.${css.header}`).first().prop('className');
+		const header = screen.getByTestId('header');
 
-		expect(actual).toContain(expected);
+		expect(header).toHaveClass(expected);
```
Comment (alexandrumorariu) [inline Panels/tests/Header-specs.js:35]: This line also.

File: Panels/tests/Header-specs.js [hunk 3]
```diff
@@ -1,66 +1,54 @@
...
-			<Header centered><title>Centered Header</title></Header>
-		);
+		render(<Header data-testid="header" centered><title>Centered Header</title></Header>);
 
 		const expected = css.centered;
-		const actual = subject.find(`.${css.header}`).first().prop('className');
+		const header = screen.getByTestId('header');
 
-		expect(actual).toContain(expected);
+		expect(header).toHaveClass(expected);
```
Comment (alexandrumorariu) [inline Panels/tests/Header-specs.js:45]: This line also.

File: Panels/tests/Header-specs.js [hunk 4]
```diff
@@ -103,54 +91,41 @@ describe('Header Specs', () => {
...
-			<Header onClose={handleClose} />
-		);
+		render(<Header onClose={handleClose} />);
 
-		tap(subject.find(`.${css.slotAfter}`).find('Button'));
+		fireEvent.mouseDown(screen.getByRole('button'));
+		fireEvent.mouseUp(screen.getByRole('button'));
 
 		const expected = 1;
 		const actual = handleClose.mock.calls.length;
```
Comment (alexandrumorariu) [inline Panels/tests/Header-specs.js:131]: Here you can replace it with toHaveBeenCalled(), but it's your choice.

**Review by alexandrumorariu — APPROVED** [review]
LGTM

File: Panels/tests/Header-specs.js [hunk 5]
```diff
@@ -103,105 +88,53 @@ describe('Header Specs', () => {
...
-			<Header onClose={handleClose} />
-		);
-
-		tap(subject.find(`.${css.slotAfter}`).find('Button'));
+		render(<Header onClose={handleClose} />);
 
-		const expected = 1;
-		const actual = handleClose.mock.calls.length;
+		fireEvent.mouseDown(screen.getByRole('button'));
+		fireEvent.mouseUp(screen.getByRole('button'));
```
Comment (stanca-pop-lgp) [inline Panels/tests/Header-specs.js:125]: Is there a reason why you used these 2 lines and not `userEvent.click(screen.getByRole('button'))`?

File: Panels/tests/Header-specs.js [hunk 6]
```diff
@@ -103,105 +88,53 @@ describe('Header Specs', () => {
...
-		const closeButton = subject.find(`.${css.slotAfter}`).find('Button');
-		const expected = 0;
-		const actual = closeButton.length;
+		const closeButton = screen.queryByLabelText('Exit app');
 
-		expect(actual).toBe(expected);
+		expect(closeButton).toBeNull();
 	});
 
 	test('should call onClose when close button is clicked', () => {
```
Comment (stanca-pop-lgp) [inline Panels/tests/Header-specs.js:120]: You can use async to get rid of the console warning.
```suggestion
	test('should call onClose when close button is clicked', async () => {
		const handleClose = jest.fn();
		render(<Header onClose={handleClose} />);

		userEvent.click(screen.getByRole('button'));

		await waitFor(() =>
			expect(handleClose).toHaveBeenCalled()
		)
        });
```

File: Panels/tests/Header-specs.js (hunk 5 above)
Comment (daniel-stoian-lgp) [reply Panels/tests/Header-specs.js:125]: With userEvent.click, onTap will be called twice

image

File: Panels/tests/Header-specs.js [hunk 7]
```diff
@@ -69,15 +54,15 @@ describe('Header Specs', () => {
 			</Header>
 		);
 
-		const actual = subject.find(`.${css.slotAbove}`).first().text();
+		const slotAbove = screen.getByText(expected);
 
-		expect(actual).toBe(expected);
+		expect(slotAbove).toHaveClass(css.slotAbove);
```
Comment (stanca-pop-lgp) [inline Panels/tests/Header-specs.js:59]: Maybe create `expected` variables like above (for css classes) for consistency. For the others below as well.

File: Panels/tests/Panels-specs.js [hunk 8]
```diff
@@ -1,297 +1,251 @@
...
+			);
 
-			expect(actual).toBe(expected);
-		}
-	);
+			const backButton = screen.getByLabelText(label);
+			const expectedClass = 'slotBefore';
+
+			expect(backButton).toBeInTheDocument();
+			expect(backButton.parentElement.parentElement).toHaveClass(expectedClass);
```
Comment (stanca-pop-lgp) [inline Panels/tests/Panels-specs.js:181]: You could create a `backButtonContainer` (or so) variable for backButton.parentElement.parentElement. For all the below as well.

**Review by stanca-pop-lgp — APPROVED** [review]

**Pull request merged**